src/Compilers/CSharp/Portable/BoundTree/BoundExpressionExtensions.cs

```diff
@@ -175,5 +176,67 @@ internal static bool IsExpressionOfComImportType(this BoundExpression expression
             TypeSymbol receiverType = expressionOpt.Type;
             return (object)receiverType != null && receiverType.Kind == SymbolKind.NamedType && ((NamedTypeSymbol)receiverType).IsComImport;
         }
+
+        internal static TypeSymbolWithAnnotations GetTypeAndNullability(this BoundExpression expr, bool includeNullability)
+        {
+            var type = expr.Type;
+            if ((object)type == null)
+            {
+                return null;
+            }
+            // PROTOTYPE(NullableReferenceTypes): Could we track nullability always,
+            // even in C#7, but only report warnings when the feature is enabled?
+            var isNullable = includeNullability ? expr.IsNullable() : null;
+            return TypeSymbolWithAnnotations.Create(type, isNullable);
+        }
+
+        internal static bool? IsNullable(this BoundExpression expr)
```

Comment:
>IsNullable [](start = 30, length = 10)

It doesn't look like this method takes into account any suppression for nullable modifiers except the SuppressNullableWarningExpression (assembly/symbol level). Have we decided to not support any of those suppressions/opt out mechanisms?

Reply:
For example binder has helpers like GetFieldTypeWithAdjustedNullableAnnotations, etc.

---
In reply to: [139053137](https://github.com/dotnet/roslyn/pull/21967#discussion_r139053137) [](ancestors = 139053137)